### PR TITLE
[Java] Fix out-dated signatures of JNI methods

### PR DESCRIPTION
**Pull request by jovany-wang**
## What do these changes do?
When Java code invoke native code by JNI, it will find the function by the function name which is declared in native code(`org_ray_spi_impl_DefaultLocalSchedulerClient.cc`). If there are some overloading functions, the JNI will find the most matching one by the `signature` in the comment, like this:
```c++
/*
 * Class:     org_ray_spi_impl_DefaultLocalSchedulerClient
 * Method:    _reconstructObjects
 * Signature: (J[BZ)V   /******This is the Signature declaration.******/
 */
```
Many signatures are not correct yet. 
But because there are no overloading, this issue is not triggered.

In this PR, I do these things:
1) Renamed the native JNI methods and some parameters of JNI methods. 
2) Fixed native JNI methods' signatures by `javah` tool.
3) Removed some useless native methods.

## Related issue number
N/A

### CODE REVIEW

**Review by raulchen — COMMENTED** [review]
Thanks, looks good to me.

Most of the function parameters in `org_ray_spi_impl_DefaultLocalSchedulerClient.cc` don't have meaningful names. Could you also improve that with this PR? thanks.

**Review by robertnishihara — APPROVED** [review]
Looks good to me, though we should still get rid of the autogenerated files at some point.

**Comment by jovan-wong** [comment]
I have updated some code.

In conclusion, it contains 3 changes:
1) Renamed the native JNI methods and some parameters of JNI methods. 
2) Fixed native JNI methods' signatures by `javah` tool.
3) Removed some useless native methods.

@robertnishihara  @raulchen  Could u plz review this again?

**Comment by jovan-wong** [comment]
@eric-jj cc

**Review by raulchen — CHANGES_REQUESTED** [review]
There's lint error in `org_ray_spi_impl_DefaultLocalSchedulerClient.cc`. Otherwise looks good.

**Review by raulchen — APPROVED** [review]

**Comment by jovan-wong** [comment]
I didn't change any py file, and just changed `org_ray_spi_impl_DefaultLocalSchedulerClient.h` and `org_ray_spi_impl_DefaultLocalSchedulerClient.cc` which are used for Java. 
But in many times, the CI job `#1` and `#2` can't pass.

```shell
=============== 1 failed, 47 passed, 8 skipped in 218.85 seconds ===============
The command "python -m pytest -v test/actor_test.py" exited with 1.
```
@robertnishihara  Do u guys have any idea about it?

File: src/local_scheduler/lib/java/org_ray_spi_impl_DefaultLocalSchedulerClient.cc [hunk 1]
```diff
@@ -153,7 +153,8 @@ Java_org_ray_spi_impl_DefaultLocalSchedulerClient_nativeReconstructObjects(
   std::vector<ObjectID> object_ids;
   auto len = env->GetArrayLength(objectIds);
   for (int i = 0; i < len; i++) {
-    jbyteArray object_id_bytes = (jbyteArray) env->GetObjectArrayElement(objectIds, i);
+    jbyteArray object_id_bytes =
+        (jbyteArray) env->GetObjectArrayElement(objectIds, i);
```
Comment (robertnishihara) [inline src/local_scheduler/lib/java/org_ray_spi_impl_DefaultLocalSchedulerClient.cc]: We should not use C style casts here

File: src/local_scheduler/lib/java/org_ray_spi_impl_DefaultLocalSchedulerClient.cc [hunk 2]
```diff
@@ -209,7 +211,8 @@ Java_org_ray_spi_impl_DefaultLocalSchedulerClient_nativeWaitObject(
...
```
Comment (robertnishihara) [inline src/local_scheduler/lib/java/org_ray_spi_impl_DefaultLocalSchedulerClient.cc]: C style cast

**Comment by robertnishihara** [comment]
@jovan-wong I think those test failures are unrelated. Can you remove the C-style casts? Then I think we can merge it.

**Pull request merged**